src/main/java/org/jenkinsci/plugins/workflow/log/OutputStreamTaskListener.java

```diff
@@ -63,7 +65,7 @@ public interface OutputStreamTaskListener extends TaskListener {
             return ((OutputStreamTaskListener) listener).getOutputStream();
         }
         PrintStream ps = listener.getLogger();
-        if (ps.getClass() != PrintStream.class) {
+        if (ps.getClass() != PrintStream.class && ps.getClass() != TeePrintStream.class) {
```

Comment:
The changes here don't seem right. I think instead we should handle `TeeTaskListener` and/or `TeePrintStream` directly and have it return a `TeeOutputStream` where possible, using this method recursively as needed.